AlphaPrint.py

```diff
@@ -1,174 +1,198 @@
-                wht1 = wht1+" "
-            for k in range(24-len2):
-                wht2 = wht2+" "        
-            print("000000"+str(i)+"0"+"  "+hexed[counter:counter+24]+wht1+"  "+hexed[counter+24:counter+48]+wht2+"  "+printable[counter2:counter2+16])
-
-try:
-    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM) #open connection
-    s.connect((hostname, port))
-    print("Connected to "+ str(hostname)+ " on port "+str(port))
-except socket.error:
-    print("Connection failed, retrying..")
-    time.sleep(2) #in case something else was interfering with port, read it might cause issues
-    try: #tries again as above
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        s.connect((hostname, port))
-        print("Connected to "+ str(hostname)+ " on port "+str(port))
-    except socket.error:
-        print()
-        print("Connection to "+ str(hostname)+ " on port "+ str(port)+ " failed")
-        sys.exit(0)
-
-#s.send(bytes("\002Test\n","utf-8"))
-#MAY NOT BE NEEDED NOW
-
-if filename != "AlphaFile.txt": #if they specified a file
-    f = open(filename, "rb")
-    l = f.read()
-
-    #length of file. I think this is better than len() for this implementation
-    f.seek(0, 2)
-    length = f.tell()
-    f.seek(0, 0)
-else: #If they didn't default file contents
-    contents = "THIS IS A TEST\012"
-    length = len(contents)*60
-
-
-s.send("\002"+printer+"\012".encode("utf-8")) #establish printer
-print("Establishing printer...")
-hexdump("\002"+printer+"\012")
-ack()
-
-
-#compile control file--------------------------------
-control = 'H'+hostname+"\012"+'N'+filename+"\012"+'P'+user+"\012"+'o'+filename+"\012"
-s.send("\002"+str(len(control))+" cfA000"+hostname+"\012".encode("utf-8"))
-
-print("Assembling control file...")
-hexdump("\002"+str(len(control))+" cfA000"+hostname+"\012")
-ack()
-
-print("Sending control file...")
-hexdump(control)
-s.send(control.encode("utf-8"))
-
-s.send("\000".encode("utf-8"))
-
-ack()
-#end of control file
-
-
-#compile data file--------------------------------
-data = "\003"+str(length)+" dfA000"+hostname+"\012"
-s.send(data.encode("utf-8"))
-print("Assembling and sending data file...")
-hexdump(data)
-
-ack()
-#end of data file
-
-
-#Sending data--------------------------------
-print("Sending print job...")
-
-if filename!="AlphaFile.txt": #send their file
-    while True:
-        contents = f.readline()
-        if not contents:
-            break
-        s.send(contents)
-else: #send a the default file
-    for i in range(60):
-        s.send(contents.encode("utf-8"))
-
-s.send("\000".encode("utf-8"))
-
-print("Closing connection...")
-s.close()
+#!/usr/bin/env python
+
+#Jazz Sussman Moss
+#6/25/2020
+
+#imports
+import socket, time, os, getpass, string, sys, argparse
+
+parser = argparse.ArgumentParser(description="defaults")
+parser.add_argument('-v', action='store_true', help='Use to enable verbose mode. Displays hexdump and some step-by-step confirmation')

```

Comment:
Let's eliminate conversational text. e.g. ...help='Produce verbose output. Displays hexdump and step-by-step confirmation'. I usually look at man pages on a mac like tar or other network utilities for inspiration. Let's change "Don't specify for default" to "default is ..." where you explicitly state the default value.